[build-system]
requires = ["setuptools>=65.0.0", "wheel", "setuptools-git-versioning"]
build-backend = "setuptools.build_meta"

[tool.setuptools.packages]
find = {namespaces = false}

[tool.setuptools-git-versioning]
enabled = true

[project]
name = "choreographer"
description = "Devtools Protocol implementation for chrome."
readme = "README.md"
requires-python = ">=3.9"
dynamic = ["version"]
authors = [
  {name = "Andrew Pikul", email="ajpikul@gmail.com"},
  {name = "Neyberson Atencio", email="neyberatencio@gmail.com"}
  ]
maintainers = [
  {name = "Andrew Pikul", email = "ajpikul@gmail.com"},
]
dependencies = [
  "simplejson"
  ]

[project.optional-dependencies]
dev = [
  "pytest",
  "pytest-asyncio",
  "pytest-xdist",
  "async-timeout",
  "poethepoet>=0.31.1",
]

[project.scripts]
dtdoctor = "choreographer.browser:diagnose"

[tool.ruff.lint]
ignore = ["E701"]

[tool.pytest.ini_options]
asyncio_default_fixture_loop_scope = "function"

[tool.poe.tasks]
# the capture on these is weird, the mechanics are weird, i forget exactly whats its doing

_test_proc = "pytest -W error -n auto -v -rfE --capture=fd tests/test_process.py"
_test_fn = "pytest -W error -n auto -v -rfE --ignore=tests/test_process.py"
# runs all the tests quickly
test = ["_test_proc", "_test_fn"]

_debug-test_proc = "pytest -W error -vvx -rA --capture=tee-sys tests/test_process.py"
_debug-test_fn = "pytest -W error -vvvx -rA --capture=tee-sys --ignore=tests/test_process.py"
# runs the tests one and a time and stops on error
debug-test = ["_debug-test_proc", "_debug-test_fn"]
# run test with basic setting but expect file/filter from you
filter-test = "pytest -W error -vvvx -rA --capture=tee-sys"
